app/javascript/components/Study.js

```diff
@@ -1,57 +1,103 @@
+    if (amountOfMatchesInSummaryWordLimit>0) {
+      //  Need to recaluculate index positions because added html changes size of textDescription
+      const beginningTextIndex= (amountOfMatchesInSummaryWordLimit *(lengthOfHighlightTag+term.length))
+      const displayedBeginningText = { __html: styledText.slice(0, beginningTextIndex+summaryWordLimit) }
+      return <><span dangerouslySetInnerHTML={displayedBeginningText}></span> <span className="detail">... </span><span dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}</>
+    }
+    const displayedBeginningText = styledText.slice(0, summaryWordLimit)
+    return <><span>{displayedBeginningText} </span><span className="detail">... </span> <span dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}</>
+  }
+  const displayedStudyDescription = { __html: styledText.slice(0, descriptionWordLimit) }
+  if (textDescription.length>descriptionWordLimit) {
+    return <><span dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}</>
+  } else {
+    return <><span dangerouslySetInnerHTML={displayedStudyDescription}></span></>
   }
-  return <span>{shortenedText}{suffixTag}</span>
 }
 
 /* removes html tags from a string */
 function stripTags(rawString) {
   const tempDiv = document.createElement('div')
-  // Set the HTML content with the providen
+  // Set the HTML content with the provided
   tempDiv.innerHTML = rawString
   // Retrieve the text property of the element
   return tempDiv.textContent || ''
 }
 
 /* displays a brief summary of a study, with a link to the study page */
-export default function Study({study}) {
-  const studyDescription = formatDescription(study.description)
+export default function Study({ study }) {
+  const { terms, facets } = study
+  const studyTitle= highlightText(study.name, terms).styledText
+  const studyDescription = formatDescription(study.description, terms)
+  const displayStudyTitle = { __html: studyTitle }
   let inferredBadge = <></>
   if (study.inferred_match) {
     const helpText = `${study.term_matches.join(', ')} was not found in study metadata, only in study title or description`
-    inferredBadge = <span className="badge soft-badge"  data-toggle="tooltip" title={helpText}>text match only</span>
+    inferredBadge = <span className="badge soft-badge" data-toggle="tooltip" title={helpText}>text match only</span>
   }
 
+
   return (
     <>
-      <label htmlFor={study.name} id= 'result-title'>
-        <a href={study.study_url}>{study.name} </a> {inferredBadge}</label>
-      <div>
-        <span id='cell-count' className='badge badge-secondary'>
-          {study.cell_count} Cells
-        </span>
+      <div key={study.accession}>
+        <label htmlFor={study.name} id= 'result-title'>
+          <a href={study.study_url} dangerouslySetInnerHTML = {displayStudyTitle}></a>{inferredBadge}
+        </label>
+        <div>
+          <span id='cell-count' className='badge badge-secondary'>
```

Comment:
there will be many spans like this, so we shouldn't use 'id' -- put 'cell-count' as a class